addons/website_sale/models/website.py

```diff
@@ -516,6 +519,33 @@ def _get_product_page_grid_image_classes(self):
         return spacing_map.get(self.product_page_image_spacing) + ' ' +\
                 columns_map.get(self.product_page_grid_columns)
 
+    @api.model
+    def _send_abandoned_cart_email(self):
+        for website in self.search([]):
+            if not website.send_abandoned_cart_email:
+                continue
+            SaleOrder = self.env['sale.order']
+            abandoned_carts = SaleOrder.search([
+                ('is_abandoned_cart', '=', True),
+                ('cart_recovery_email_sent', '=', False)
+            ])
```

Comment:
You are iterating through the websites so the sales orders you fetch should only be the one associated with the website.
```suggestion
            SaleOrder = self.env['sale.order']
            abandoned_carts = SaleOrder.search([
                ('is_abandoned_cart', '=', True),
                ('cart_recovery_email_sent', '=', False),
                ('website_id', '=', website.id),
            ])
```